### PR TITLE
More Docstrings

### PR DESCRIPTION
**Pull request by jamesmcclain**
## Overview

More docstrings.  All of the `with_*` methods in the [public API](https://docs.rastervision.io/en/rde-feature-docs/api.html) should now be covered.

### Checklist

- [x] Ran scripts/format_code and commited any changes
- [x] Documentation updated if needed
- [x] PR has a name that won't get you publicly shamed for vagueness

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* How to test this PR
* Prefer bulleted description
* Start after checking out this branch
* Include any setup required, such as rebuilding the Docker image.
* Include test case, and expected output if not captured by automated tests.

Closes #XXX

### CODE REVIEW

File: rastervision/data/raster_source/image_source_config.py [hunk 1]
```diff
@@ -82,6 +82,7 @@ def from_proto(self, msg):
             .with_uri(msg.image_file.uri)
 
     def with_uri(self, uri):
+        """Set URI for a GeoTIFF used to read labels."""
```
Comment (lewfish) [inline rastervision/data/raster_source/image_source_config.py]: ImageSource is for non-georeferenced image files including tifs and pngs. not sure if rasterio can open jpgs.

Comment (lewfish) [reply rastervision/data/raster_source/image_source_config.py]: Also, this is used to read raster data, not necessarily labels.

File: rastervision/data/raster_source/geotiff_source_config.py [hunk 2]
```diff
@@ -84,11 +84,13 @@ def from_proto(self, msg):
             .with_uris(msg.geotiff_files.uris)
 
     def with_uris(self, uris):
+        """Set URIs for a GeoTIFFs used to read labels."""
```
Comment (lewfish) [inline rastervision/data/raster_source/geotiff_source_config.py]: same comment about not reading labels

File: rastervision/command/train_command_config.py [hunk 3]
```diff
@@ -92,6 +92,7 @@ def with_experiment(self, experiment_config):
         return b
 
     def with_task(self, task):
+        """Sets a specific task type, e.g. rv.OBJECT_DETECTION."""
```
Comment (lewfish) [inline rastervision/command/train_command_config.py]: This makes it seem like you can pass rv.OBJECT_DETECTION, but you actually need to pass a TaskConfig object. same comment applies in other places.

**Comment by jamesmcclain** [comment]
Changes in progress.

File: rastervision/data/raster_source/image_source_config.py (hunk 1 above)
Comment (jamesmcclain) [reply rastervision/data/raster_source/image_source_config.py]: Done.

File: rastervision/data/raster_source/geotiff_source_config.py (hunk 2 above)
Comment (jamesmcclain) [reply rastervision/data/raster_source/geotiff_source_config.py]: Done.

File: rastervision/command/train_command_config.py (hunk 3 above)
Comment (jamesmcclain) [reply rastervision/command/train_command_config.py]: Done.

**Pull request merged**